src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -70,5 +89,25 @@ private static Architecture Map(int processorArchitecture)
                     return Architecture.X86;
             }
         }
+
+        private static Architecture MapMachineConstant(ushort processMachine)
+        {
+            switch (processMachine)
+            {
+                case 0x01C0: // IMAGE_FILE_MACHINE_ARM
+                case 0x01C2: // IMAGE_FILE_MACHINE_THUMB
+                case 0x01C4: // IMAGE_FILE_MACHINE_ARMNT
+                    return Architecture.Arm;
+
+                case 0x8664: // IMAGE_FILE_MACHINE_AMD64
+                    return Architecture.X64;
+
+                case 0xAA64: // IMAGE_FILE_MACHINE_ARM64
+                    return Architecture.Arm64;
+
+                default: // IMAGE_FILE_MACHINE_UNKNOWN etc.
+                    return ProcessArchitecture();
```

Comment:
We do not expect to take this path. We should assert here.

Reply:
I must have misread your comment  about using ProcessArchitecutre as a fallback. Is that what you want to keep?